src/controllers/UserController.java

```diff
@@ -92,7 +96,8 @@ private Command ContactDetails() {
         UserController<?> currentController = this;
         ContactManager contactManager = new ContactManager(getDatabase());
         ContactData contactData = contactManager.getContactData(userData);
-        ContactController contactController = new ContactController(getContext(), currentController, contactData);
+        ContactController contactController = new ContactController(getContext(), currentController, contactData,
```

Comment:
why is it taking in contact data if it can derive contact data from Userdata

Reply:
There currently isn't a method to get the contact data from the user data, but i will implement it in the next push and change the code accordingly.